### DISCUSSION TITLE
Problem: Using View Transition API requires switching from useLazyAsyncData to useAsyncData which causes a delay

### DISCUSSION THREAD

**Discussion by toddpadwick** [nuxt/nuxt#26200] [Questions]
For a few specific page changes on my website, I am opting for the [View Transitions API](https://developer.mozilla.org/en-US/docs/Web/API/View_Transitions_API), newly introduced to Chromium and introduced as an experimental feature in Nuxt 3. To see these transitions, scroll down to the case studies on [this homepage](https://www.functionandform.co.uk/) in **Chrome**, and click on them to see them seamlessly transition to becoming the case study page.

https://github.com/nuxt/nuxt/assets/36191462/5d86842b-b11a-4f41-9357-9752bd0392e6

to get View Transition API to work with Nuxt, I had to switch the case study page queries from `useAsyncData` to `useLazyAsync` data. My understanding of the differences between these two are:
- **[lazyAsyncData](https://nuxt.com/docs/api/composables/use-lazy-async-data)** allows the page template to change before or simaltanous to the actual query meaning the content can be loaded dynamically - this creates a better UX for the user since they see instant change in state, whether the content has even loaded or not.
- **[asyncData](https://nuxt.com/docs/api/composables/use-async-data)** does absolutely nothing and the DOM remains as is, until the query has returned a result, meaning the user has to wait for any visible change to occur until content has loaded in its entirety. obviously, this is not ideal as even the slightest delay, feels sluggish and looks like the users link click didn't work.

Because View Transitions API requires the subsequent pages' DOM content to be available at the moment the page changes, so that it can match up the `view-transition-name` elements, LazyAsyncData does not work. This, of course, is not ideal: as you'll see when clicking on the case studies on [my homepage](https://www.functionandform.co.uk/), there is a lag. So my question to the Nuxt community is whether there is a better way of doing this to keep the View Transition without the resulting delay?

**Comment by charlesoestergaard** [comment]
I have the exact same problem. Hope someone has the answer...
